Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -39,144 +39,210 @@ inline std::ostream& operator<<(std::ostream& os, IntersectionStatus status) {
+      : m_position(position), m_pathLength(pathLength), m_status(status) {}
 
-  /// Default constructor
-  Intersection() = default;
+  /// Returns whether the intersection was successful or not
+  explicit operator bool() const { return m_status != Status::missed; }
 
-  /// Bool() operator for validity checking
-  explicit operator bool() const { return (status != Status::missed); }
+  const Position& position() const { return m_position; }
 
-  /// Smaller operator for sorting,
-  /// - it respects the validity of the intersection
-  /// @param si is the intersection for testing
-  bool operator<(const Intersection<DIM>& si) const {
-    if (status == Status::unreachable) {
-      return false;
-    }
-    // Now check the pathLength
-    if (si.status != Status::unreachable) {
-      return (pathLength < si.pathLength);
-    }
-    // The current one wins, no re-ordering
-    return true;
+  ActsScalar pathLength() const { return m_pathLength; }
+
+  Status status() const { return m_status; }
+
+  static bool forwardOrder(const Intersection& aIntersection,
+                           const Intersection& bIntersection) {
+    auto a = aIntersection.pathLength();
+    auto b = bIntersection.pathLength();
+    return a < b;
   }
 
-  /// Greater operator for sorting,
-  /// - it respects the validity of the intersection
-  /// @param si is the intersection for testing
-  bool operator>(const Intersection<DIM>& si) const {
-    if (status == Status::unreachable) {
+  static bool closestOrder(const Intersection& aIntersection,
+                           const Intersection& bIntersection) {
+    if ((aIntersection.status() == Status::unreachable) &&
+        (bIntersection.status() != Status::unreachable)) {
       return false;
     }
-    // Now check the pathLength
-    if (si.status != Status::unreachable) {
-      return (pathLength > si.pathLength);
+    if ((aIntersection.status() != Status::unreachable) &&
+        (bIntersection.status() == Status::unreachable)) {
+      return true;
     }
-    // The current one wins, no re-ordering
-    return true;
+    // both are reachable or onSurface now
+    auto a = aIntersection.pathLength();
+    auto b = bIntersection.pathLength();
+    return std::abs(a) < std::abs(b);
   }
+
+ private:
+  /// Position of the intersection
+  Position m_position = Position::Zero();
+  /// Signed path length to the intersection (if valid)
+  ActsScalar m_pathLength = std::numeric_limits<double>::infinity();
+  /// The Status of the intersection
+  Status m_status = Status::unreachable;
 };
 
 using Intersection2D = Intersection<2>;
-
 using Intersection3D = Intersection<3>;
 
+using MultiIntersection3D = std::array<Intersection3D, 2>;
+
 /// @brief class extensions to return also the object and a representation
 template <typename object_t, typename representation_t = object_t>
 class ObjectIntersection {
  public:
+  /// Default constructor
+  ObjectIntersection() = default;
```

Comment:
Was this default-constructible before (I guess implicitly so)? Do we need this for vectors or is there another reason?

Reply:
it was default constructible before. for some reason the diff moved here quite a bit. we need to make it explicit because the other constructor would shadow the default one

Reply:
but would be interesting to see where a default constructed Intersection is needed, and why...

Reply:
I think currently this is the mechanism for an invalid intersection but this could also be solved with `std::optional`

Reply:
Hmm if this works reliably, I would prefer a private default constructor and a factory `ObjectIntersection::Invalid()` probably...

Reply:
Personally, I think I'd prefer explicit `std::optional`, but I won't block another solution if you think it makes more sense.